scipy/stats/_qmc.py

```diff
@@ -1595,6 +1596,302 @@ def fast_forward(self, n: IntNumber) -> Sobol:
         return self
 
 
+class PoissonDisk(QMCEngine):
+    """Poisson disk sampling.
+
+    Parameters
+    ----------
+    d : int
+        Dimension of the parameter space.
+    radius : float
+        Minimal distance to keep between points when sampling new candidates.
+    hypersphere : {"volume", "surface"}, optional
+        Sampling strategy to generate potential candidates to be added in the
+        final sample. Default is "volume".
+
+        * ``volume``: original Bridson algorithm as described in [1]_.
+          new candidates are sampled within the hypersphere.
+        * ``surface``: only sample the surface of the hypersphere.
+
+    candidates : int
+        Number of candidates to sample per iteration. More candidates result
+        in a denser sampling as more candidates can be accepted per iteration.
+    seed : {None, int, `numpy.random.Generator`}, optional
+        If `seed` is None the `numpy.random.Generator` singleton is used.
+        If `seed` is an int, a new ``Generator`` instance is used,
+        seeded with `seed`.
+        If `seed` is already a ``Generator`` instance then that instance is
+        used.
+
+    Notes
+    -----
+
+    Poisson disk sampling is an iterative sampling strategy. Starting from
+    a seed sample, ``n`` `candidates` are sampled in the hypersphere
+    surrounding the seed. Candidates bellow a certain `radius` or outside the
+    domain are rejected. New samples are added in a pool of sample seed. The
+    process stops when the pool is empty or when the number of required
+    samples is reached.
+
+    The maximum number of point that a sample can contain is directly linked
+    to the `radius`. As the dimension of the space increases, a higher radius
+    spread the points further and help overcome the curse-of-dimensionality.
+
+    .. warning::
+
+       The algorithm is more suitable for low dimensions and sampling size
+       due to its iterative nature and memory requirements.
+
+    Some code taken from [2]_ by P. Zun, written consent given on 31.03.2021
```

Comment:
P. Zun (@diregoblin) got the code from [2] (actually he tracked down some Matlab code and it lead to this SO post). There is more detail in the long discussion we had on this PR at the beginning.

Reply:
@diregoblin presumably already gave permission, since this is their PR. So, should be enough to simply state that you had received written permission from StackOverflow user Shamis and link to the S/O URL as you've done.

Reply:
Sure I can remove the precision on who took the code on SO.